pkg/tsdb/prometheus/querydata/exemplar_sampler.go

```diff
@@ -98,6 +100,15 @@ func (e *exemplarSampler) getLabelNames() []string {
 
 // getSampledExemplars returns the exemplars sorted by timestamp
 func (e *exemplarSampler) getSampledExemplars() []exemplar {
+	if e.disableSampling {
+		var exemplars []exemplar
+		for _, bucket := range e.buckets {
+			exemplars = append(exemplars, bucket...)
+		}
+
+		return exemplars
+	}
+
 	exemplars := make([]exemplar, 0, len(e.buckets))
```

Comment:
Can't we move this line to line 103 and we do not need to define it again inside `if`?

Reply:
Refactored a bit following our discussion